// @vechain/sdk-ethers-adapter/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  adapters: () => ethers_adapter_exports,
  contractAdapter: () => contractAdapter,
  factoryAdapter: () => factoryAdapter,
  helpers: () => helpers
});
module.exports = __toCommonJS(src_exports);

// src/ethers-adapter.ts
var ethers_adapter_exports = {};
__export(ethers_adapter_exports, {
  contractAdapter: () => contractAdapter,
  factoryAdapter: () => factoryAdapter,
  helpers: () => helpers
});

// src/adapters/helpers.ts
var getContractAddress = async (txId, hardhatVeChainProvider) => {
  const tx = await hardhatVeChainProvider.thorClient.transactions.waitForTransaction(
    txId
  );
  return tx?.outputs[0].contractAddress ?? "";
};
var helpers = {
  getContractAddress
};

// src/adapters/contract-adapter.ts
var contractAdapter = (contract, hardhatVeChainProvider) => {
  contract.getAddress = async function getAddress() {
    return await helpers.getContractAddress(
      contract.deploymentTransaction()?.hash ?? "",
      hardhatVeChainProvider
    );
  };
  return contract;
};

// src/adapters/factory-adapter.ts
var import_sdk_core = require("@vechain/sdk-core");
function factoryAdapter(contractFactory, hardhatVeChainProvider) {
  contractFactory.deploy = async function(...args) {
    const tx = await this.getDeployTransaction(...args);
    import_sdk_core.vechain_sdk_core_ethers.assert(
      this.runner != null && typeof this.runner.sendTransaction === "function",
      "factory runner does not support sending transactions",
      "UNSUPPORTED_OPERATION",
      {
        operation: "sendTransaction"
      }
    );
    const sentTx = await this.runner.sendTransaction(tx);
    const receipt = await hardhatVeChainProvider.thorClient.transactions.waitForTransaction(
      sentTx.hash
    );
    return new import_sdk_core.vechain_sdk_core_ethers.BaseContract(
      receipt?.outputs[0].contractAddress ?? "",
      this.interface,
      this.runner,
      sentTx
    );
  };
  return contractFactory;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  adapters,
  contractAdapter,
  factoryAdapter,
  helpers
});
